src/se/vidstige/jadb/JadbDevice.java

```diff
@@ -175,9 +160,9 @@ public void push(InputStream source, long lastModified, int mode, RemoteFile rem
     }
 
     public void push(File local, RemoteFile remote) throws IOException, JadbException {
-        FileInputStream fileStream = new FileInputStream(local);
-        push(fileStream, local.lastModified(), getMode(local), remote);
-        fileStream.close();
+        try (FileInputStream fileStream = new FileInputStream(local)) {
+            push(fileStream, local.lastModified(), DEFAULT_MODE, remote);
+        }
```

Comment:
let's keep the `getMode` function I feel. It makes it clearer that it's a function of the file object. In fact, I thin in newer versions of java you _can_ get the mode on Mac + Linux(?).

Reply:
:man_shrugging:  :-)